### PR TITLE
hw01 classes

### PR DESCRIPTION
**Pull request by OmarTayyar**
except one thing i did everthing, that was ver sly and almost sly part

### CODE REVIEW

File: src/happy_family/Human.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public String surname;
+    public Integer year;
+    public Integer IQ;
+    public Pet pet;
+    public Human mother;
+    public Human father;
+    public String[][] schedule;
+
+    public Human() {
+
```
Comment (Shammad-Kosayev) [inline src/happy_family/Human.java]: İ dont think there is a need for an empty line in empty constructor

File: src/happy_family/Human.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        this.surname = surname;
+        this.year = year;
+        this.pet = pet;
+        this.father = father;
+        this.mother = mother;
+        this.schedule = schedule;
+    }
+
+    public void greetPet() {
+        System.out.println("Hello, " + pet.nickname);
```
Comment (Shammad-Kosayev) [inline src/happy_family/Human.java]: Use 'printf' please ( dont forget '/n' in the end )

File: src/happy_family/Pet.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package happy_family;
+
+import com.sun.org.apache.xpath.internal.operations.String;
```
Comment (Shammad-Kosayev) [inline src/happy_family/Pet.java]: What is this? Wrong import i guess

File: src/happy_family/Pet.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import java.util.Arrays;
+
+public class Pet {
+    public String species;
+    public String nickname;
+    public Integer age;
+    public Integer tricklevel;
+    public String[] habits;
+
```
Comment (Shammad-Kosayev) [inline src/happy_family/Pet.java:11]: No need for another empty line

File: src/happy_family/Pet.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        this.nickname = nickname;
+    }
+
+
+    public void eat() {
+        System.out.println("I am eating");
+    }
+
+    public void respond() {
+        System.out.println("Hello, owner. I am " + nickname + "I miss you");
```
Comment (Shammad-Kosayev) [inline src/happy_family/Pet.java]: Use 'printf' please ( dont forget '/n' in the end )

File: src/happy_family/Pet.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public void foul() {
+        System.out.println("I need to cover it up");
+    }
+
+    public java.lang.String toString() {
```
Comment (Shammad-Kosayev) [inline src/happy_family/Pet.java]: Fix this please. Use normal string

File: src/happy_family/Pet.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        return "nickname = " + nickname
```
Comment (Shammad-Kosayev) [inline src/happy_family/Pet.java]: 1) Return string is wrong. Read documentation again. (toString of Pet class)
2) use "".formatted or String.format()

**Review by Shammad-Kosayev — CHANGES_REQUESTED** [review]
Please do changes

File: src/happy_family/Human.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
+package happy_family;
+
+public class Human {
```
Comment (ayten-github) [inline src/happy_family/Human.java:3]: must be a beak line,pls

File: src/happy_family/Human.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    public void greetPet() {
+        System.out.println("Hello, " + pet.nickname);
+    }
+
+    public void describePet() {
+        System.out.printf("I have an %sis %dyears old, he is%d%n", pet.species, pet.age, pet.tricklevel);
+    }
+
+    public String toString() {
+        return String.format("Human{name = %s,surname = %s,year = %d,iq= %d,mother = %s%s,%s", name, surname, year, IQ, mother.name, mother.surname, pet.toString());
```
Comment (ayten-github) [inline src/happy_family/Human.java]: write this in 3line pls

File: src/happy_family/Pet.java (hunk 7 above)
Comment (ayten-github) [inline src/happy_family/Pet.java]: use just String.format()

File: src/happy_family/Pet.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
+package happy_family;
+
+import com.sun.org.apache.xpath.internal.operations.String;
+
+import java.util.Arrays;
+
+public class Pet {
```
Comment (ayten-github) [inline src/happy_family/Pet.java:5]: Same, break line here pls

**Pull request merged**